server.go

```diff
@@ -66,12 +66,18 @@ func Accept(w http.ResponseWriter, r *http.Request) (*Conn, error) {
 type flushWrite struct {
 	w io.Writer
 	f http.Flusher
+	r *http.Request
 }
 
 func (w *flushWrite) Write(data []byte) (int, error) {
-	n, err := w.w.Write(data)
-	w.f.Flush()
-	return n, err
+	select {
+	case <-w.r.Context().Done():
```

Comment:
Insded of `<-w.r.Context().Done()` in a `select` statement, you can check the `e.r.Context().Err()` value, which will make the code simpler and clearer.